--- a/src/test/java/org/bukkit/craftbukkit/legacy/MaterialReroutingTest.java
+++ b/src/test/java/org/bukkit/craftbukkit/legacy/MaterialReroutingTest.java
@@ -55,6 +_,7 @@
                 .filter(entry -> !entry.getName().endsWith("BlockType.class"))
                 .filter(entry -> !entry.getName().endsWith("ItemType.class"))
                 .filter(entry -> !entry.getName().endsWith("Registry.class"))
+                .filter(entry -> !entry.getName().endsWith("BaseItemBuilder.class"))
                 .filter(entry -> !entry.getName().startsWith("org/bukkit/material"))
                 // Paper start - types that cannot be translated to ItemType/BlockType
                 .filter(entry -> !entry.getName().equals("com/destroystokyo/paper/MaterialSetTag.class"))
