--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockType.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockType.java
@@ -3,9 +_,7 @@
 import com.google.common.base.Preconditions;
 import java.lang.reflect.Method;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.function.Consumer;
-import com.google.common.collect.ImmutableList;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.world.InteractionHand;
@@ -51,7 +_,7 @@
     }
 
     public static BlockType minecraftToBukkitNew(Block minecraft) {
-        return CraftRegistry.minecraftToBukkit(minecraft, Registries.BLOCK);
+        return CraftRegistry.minecraftToBukkit(minecraft, Registries.BLOCK, Registry.BLOCK);
     }
 
     public static Block bukkitToMinecraftNew(BlockType bukkit) {
@@ -147,16 +_,6 @@
     @Override
     public B createBlockData() {
         return this.createBlockData((String) null);
-    }
-
-    @Override
-    public @NotNull Collection<B> createBlockDataStates() {
-        final ImmutableList<BlockState> possibleStates = this.block.getStateDefinition().getPossibleStates();
-        final ImmutableList.Builder<B> builder = ImmutableList.builderWithExpectedSize(possibleStates.size());
-        for (final BlockState possibleState : possibleStates) {
-            builder.add(this.blockDataClass.cast(possibleState.createCraftBlockData()));
-        }
-        return builder.build();
     }
 
     @Override
