--- a/src/main/java/io/papermc/paper/registry/PaperRegistryBuilder.java
+++ b/src/main/java/io/papermc/paper/registry/PaperRegistryBuilder.java
@@ -12,8 +_,14 @@
 
         B fill(Conversions conversions, @Nullable M nms);
 
-        default B create(final Conversions conversions) {
-            return this.fill(conversions, null);
+        default Factory<M, T, B> asFactory() {
+            return (lookup) -> this.fill(lookup, null);
         }
+    }
+
+    @FunctionalInterface
+    interface Factory<M, T, B extends PaperRegistryBuilder<M, T>> {
+
+        B create(Conversions conversions);
     }
 }
