--- a/src/test/java/org/bukkit/support/DummyServerHelper.java
+++ b/src/test/java/org/bukkit/support/DummyServerHelper.java
@@ -91,7 +_,7 @@
         when(instance.getPluginManager()).thenReturn(pluginManager);
         // Paper end - testing additions
 
-        io.papermc.paper.configuration.GlobalConfigTestingBase.setupGlobalConfigForTest(RegistryHelper.getRegistry()); // Paper - configuration files - setup global configuration test base
+        io.papermc.paper.configuration.GlobalConfigTestingBase.setupGlobalConfigForTest(); // Paper - configuration files - setup global configuration test base
 
         // Paper start - add test for recipe conversion
         when(instance.recipeIterator()).thenAnswer(ignored ->
