--- a/src/test/java/org/bukkit/support/RegistryHelper.java
+++ b/src/test/java/org/bukkit/support/RegistryHelper.java
@@ -75,6 +_,7 @@
     }
 
     public static void setup(FeatureFlagSet featureFlagSet) {
+        System.setProperty("Paper.pushPaperAssetsRoot", "true"); // Paper - build system changes - push asset root
         SharedConstants.tryDetectVersion();
         Bootstrap.bootStrap();
 
