--- a/src/main/java/org/bukkit/craftbukkit/map/CraftMapCursor.java
+++ b/src/main/java/org/bukkit/craftbukkit/map/CraftMapCursor.java
@@ -5,6 +_,7 @@
 import net.minecraft.core.registries.Registries;
 import net.minecraft.world.level.saveddata.maps.MapDecorationType;
 import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
 import org.bukkit.craftbukkit.CraftRegistry;
 import org.bukkit.craftbukkit.util.Handleable;
 import org.bukkit.map.MapCursor;
@@ -16,7 +_,7 @@
         private static int count = 0;
 
         public static MapCursor.Type minecraftToBukkit(MapDecorationType minecraft) {
-            return CraftRegistry.minecraftToBukkit(minecraft, Registries.MAP_DECORATION_TYPE);
+            return CraftRegistry.minecraftToBukkit(minecraft, Registries.MAP_DECORATION_TYPE, Registry.MAP_DECORATION_TYPE);
         }
 
         public static MapCursor.Type minecraftHolderToBukkit(Holder<MapDecorationType> minecraft) {
