--- a/src/main/java/io/papermc/paper/registry/legacy/DelayedRegistry.java
+++ b/src/main/java/io/papermc/paper/registry/legacy/DelayedRegistry.java
@@ -2,7 +_,6 @@
 
 import io.papermc.paper.registry.tag.Tag;
 import io.papermc.paper.registry.tag.TagKey;
-import java.util.Collection;
 import java.util.Iterator;
 import java.util.function.Supplier;
 import java.util.stream.Stream;
@@ -40,6 +_,11 @@
     }
 
     @Override
+    public T getOrThrow(final NamespacedKey key) {
+        return this.delegate().getOrThrow(key);
+    }
+
+    @Override
     public Iterator<T> iterator() {
         return this.delegate().iterator();
     }
@@ -62,10 +_,5 @@
     @Override
     public @NonNull Tag<T> getTag(final TagKey<T> key) {
         return this.delegate().getTag(key);
-    }
-
-    @Override
-    public Collection<Tag<T>> getTags() {
-        return this.delegate().getTags();
     }
 }
