--- a/src/main/java/io/papermc/paper/brigadier/NullCommandSender.java
+++ b/src/main/java/io/papermc/paper/brigadier/NullCommandSender.java
@@ -2,6 +_,7 @@
 
 import java.util.Set;
 import java.util.UUID;
+import de.varilx.core.common.translation.Locale;
 import net.kyori.adventure.text.Component;
 import net.md_5.bungee.api.chat.BaseComponent;
 import org.bukkit.Bukkit;
@@ -24,6 +_,22 @@
 
     private NullCommandSender() {
     }
+
+
+    // Varilx - Start
+
+
+    @Override
+    public void sendTranslatedMessage(final String key, final Object... args) {
+
+    }
+
+    @Override
+    public void sendTranslatedMessage(final String key, final String project, final Locale locale, final Object... args) {
+
+    }
+
+    // Varilx - End
 
     @Override
     public void sendMessage(final String message) {
