--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -9,7 +_,10 @@
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
+
+import de.varilx.core.common.translation.Locale;
 import io.papermc.paper.entity.LookAnchor;
+import net.kyori.adventure.title.Title;
 import org.bukkit.BanEntry;
 import org.bukkit.DyeColor;
 import org.bukkit.Effect;
@@ -38,6 +_,7 @@
 import org.bukkit.block.TileState;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.block.sign.Side;
+import org.bukkit.boss.BossBar;
 import org.bukkit.conversations.Conversable;
 import org.bukkit.event.block.BlockBreakEvent;
 import org.bukkit.event.block.BlockDropItemEvent;
@@ -61,6 +_,94 @@
  */
 public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginMessageRecipient, net.kyori.adventure.identity.Identified, net.kyori.adventure.bossbar.BossBarViewer, com.destroystokyo.paper.network.NetworkClient { // Paper
 
+    // Varilx - Start
+
+    /**
+     * Displays a translated title based on the specified title key and arguments.
+     *
+     * @param titleKey The key for the title translation.
+     * @param args The arguments to be used in the translation.
+     */
+    void showTranslatedTitle(@NotNull String titleKey, @NotNull Object... args);
+
+
+    /**
+     * Displays a translated title based on the specified title key, project, locale, and arguments.
+     *
+     * @param titleKey The key for the title translation.
+     * @param project The name of the project used for translation context.
+     * @param locale The locale to use for translation.
+     * @param args The arguments to be used in the translation.
+     */
+    void showTranslatedTitle(@NotNull String titleKey, @NotNull String project, @NotNull Locale locale, @NotNull Object... args);
+
+    /**
+     * Displays a translated title with specified times based on the title key and arguments.
+     *
+     * @param titleKey The key for the title translation.
+     * @param times The timing information for the title display.
+     * @param args The arguments to be used in the translation.
+     */
+    void showTranslatedTitle(@NotNull String titleKey, @NotNull Title.Times times, @NotNull Object... args);
+
+    /**
+     * Displays a translated title with specified times and project based on the title key and arguments.
+     *
+     * @param titleKey The key for the title translation.
+     * @param project The name of the project used for translation context.
+     * @param times The timing information for the title display.
+     * @param args The arguments to be used in the translation.
+     */
+    void showTranslatedTitle(@NotNull String titleKey, @NotNull String project, @NotNull Title.Times times, @NotNull Object... args);
+
+    /**
+     * Displays a translated title with specified times, project, and locale based on the title key and arguments.
+     *
+     * @param titleKey The key for the title translation.
+     * @param project The name of the project used for translation context.
+     * @param locale The locale to use for translation.
+     * @param times The timing information for the title display.
+     * @param args The arguments to be used in the translation.
+     */
+    void showTranslatedTitle(@NotNull String titleKey, @NotNull String project, @NotNull Locale locale, @NotNull Title.Times times, @NotNull Object... args);
+
+
+    /**
+     * Shows a translated bossbar to the user
+     * with the default Locale
+     * @param bossbarKey the message key for the bossbar
+     */
+    void showTranslatedBossbar(@NotNull String bossbarKey, @NotNull Object @NotNull ... args);
+
+    /**
+     * Shows a translated bossbar to the user
+     * with the default Locale
+     * @param bossbarKey the message key for the bossbar
+     * @param bossBar the bossbar settings
+     */
+    void showTranslatedBossbar(@NotNull String bossbarKey, @NotNull net.kyori.adventure.bossbar.BossBar bossBar, @NotNull Object @NotNull ... args);
+
+    /**
+     * Shows a translated bossbar to the user
+     * with the default Locale
+     * @param bossbarKey the message key for the bossbar
+     * @param project the project
+     * @param bossBar the bossbar settings
+     */
+    void showTranslatedBossbar(@NotNull String bossbarKey, @NotNull String project, @NotNull net.kyori.adventure.bossbar.BossBar bossBar, @NotNull Object @NotNull ... args);
+
+    /**
+     * Shows a translated bossbar to the user
+     * @param bossbarKey the message key for the bossbar
+     * @param project the project
+     * @param locale the locale
+     * @param bossBar the bossbar settings
+     * @see #showBossBar(net.kyori.adventure.bossbar.BossBar)
+     */
+    void showTranslatedBossbar(@NotNull String bossbarKey, @NotNull String project, @NotNull Locale locale, @NotNull net.kyori.adventure.bossbar.BossBar bossBar, @NotNull Object @NotNull ... args);
+
+    // Varilx - End
+
     // Paper start
     @Override
     default net.kyori.adventure.identity.@NotNull Identity identity() {
@@ -1520,6 +_,30 @@
 
         return banEntry;
     }
+
+    // Varilx - Start
+
+    /**
+     * Sends a translated action bar
+     * @param messageKey the message key
+     * @param args the args for the messag
+     * @see #sendTranslatedActionBar(String, String, Locale, Object...)
+     * @see #sendTranslatedMessage(String, String, Locale, Object...)
+     */
+    public void sendTranslatedActionBar(@NotNull String messageKey, @NotNull Object@NotNull ... args);
+
+
+    /**
+     * Sends a translated actionbar
+     * @param messageKey the message key
+     * @param project the project
+     * @param locale the locale
+     * @param args the args
+     * @see #sendTranslatedMessage(String, String, Locale, Object...)
+     */
+    public void sendTranslatedActionBar(@NotNull String messageKey, @NotNull String project, @NotNull Locale locale, Object @NotNull... args);
+
+    // Varilx - End
 
     /**
      * Sends an Action Bar message to the client.
