--- a/src/main/java/org/bukkit/craftbukkit/generator/structure/CraftStructureType.java
+++ b/src/main/java/org/bukkit/craftbukkit/generator/structure/CraftStructureType.java
@@ -2,6 +_,7 @@
 
 import net.minecraft.core.registries.Registries;
 import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
 import org.bukkit.craftbukkit.CraftRegistry;
 import org.bukkit.craftbukkit.util.Handleable;
 import org.bukkit.generator.structure.StructureType;
@@ -9,7 +_,7 @@
 public class CraftStructureType extends StructureType implements Handleable<net.minecraft.world.level.levelgen.structure.StructureType<?>> {
 
     public static StructureType minecraftToBukkit(net.minecraft.world.level.levelgen.structure.StructureType<?> minecraft) {
-        return CraftRegistry.minecraftToBukkit(minecraft, Registries.STRUCTURE_TYPE);
+        return CraftRegistry.minecraftToBukkit(minecraft, Registries.STRUCTURE_TYPE, Registry.STRUCTURE_TYPE);
     }
 
     public static net.minecraft.world.level.levelgen.structure.StructureType<?> bukkitToMinecraft(StructureType bukkit) {
