--- a/src/test/java/io/papermc/paper/adventure/AdventureCodecsTest.java
+++ b/src/test/java/io/papermc/paper/adventure/AdventureCodecsTest.java
@@ -22,7 +_,6 @@
 import net.kyori.adventure.text.event.ClickEvent;
 import net.kyori.adventure.text.event.HoverEvent;
 import net.kyori.adventure.text.format.NamedTextColor;
-import net.kyori.adventure.text.format.ShadowColor;
 import net.kyori.adventure.text.format.Style;
 import net.kyori.adventure.text.format.TextColor;
 import net.kyori.adventure.text.format.TextDecoration;
@@ -279,7 +_,6 @@
             style(TextDecoration.BOLD.withState(TextDecoration.State.NOT_SET)),
             style()
                 .font(key("kyori", "kittens"))
-                .shadowColor(ShadowColor.fromHexString("#FF00AAFF"))
                 .color(NamedTextColor.RED)
                 .decoration(TextDecoration.BOLD, true)
                 .clickEvent(openUrl("https://github.com"))
