--- a/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
+++ b/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
@@ -2,6 +_,7 @@
 
 import java.util.Set;
 import java.util.UUID;
+import de.varilx.core.common.translation.Locale;
 import org.bukkit.Bukkit;
 import org.bukkit.Server;
 import org.bukkit.command.CommandSender;
@@ -10,6 +_,7 @@
 import org.bukkit.permissions.PermissionAttachment;
 import org.bukkit.permissions.PermissionAttachmentInfo;
 import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
 
 public abstract class ServerCommandSender implements CommandSender {
     public final PermissibleBase perm; // Paper
@@ -86,6 +_,22 @@
     public Server getServer() {
         return Bukkit.getServer();
     }
+
+    // Varilx - Start
+
+
+    @Override
+    public void sendTranslatedMessage(final String key, final Object... args) {
+
+    }
+
+
+    @Override
+    public void sendTranslatedMessage(final String key, final String project, final Locale locale, final Object... args) {
+
+    }
+
+    // Varilx - End
 
     @Override
     public void sendMessage(UUID uuid, String message) {
