--- a/src/main/java/io/papermc/paper/command/brigadier/PaperCommandSourceStack.java
+++ b/src/main/java/io/papermc/paper/command/brigadier/PaperCommandSourceStack.java
@@ -1,23 +_,21 @@
 package io.papermc.paper.command.brigadier;
 
 import com.destroystokyo.paper.brigadier.BukkitBrigadierCommandSource;
-import com.google.common.base.Preconditions;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.phys.Vec2;
 import net.minecraft.world.phys.Vec3;
 import org.bukkit.Location;
 import org.bukkit.command.CommandSender;
-import org.bukkit.craftbukkit.entity.CraftEntity;
 import org.bukkit.entity.Entity;
-import org.jspecify.annotations.NonNull;
-import org.jspecify.annotations.Nullable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public interface PaperCommandSourceStack extends CommandSourceStack, BukkitBrigadierCommandSource {
 
     net.minecraft.commands.CommandSourceStack getHandle();
 
     @Override
-    default @NonNull Location getLocation() {
+    default @NotNull Location getLocation() {
         Vec2 rot = this.getHandle().getRotation();
         Vec3 pos = this.getHandle().getPosition();
         Level level = this.getHandle().getLevel();
@@ -26,7 +_,7 @@
     }
 
     @Override
-    @NonNull
+    @NotNull
     default CommandSender getSender() {
         return this.getHandle().getBukkitSender();
     }
@@ -40,12 +_,6 @@
         }
 
         return nmsEntity.getBukkitEntity();
-    }
-
-    @Override
-    default CommandSourceStack withExecutor(@NonNull Entity executor) {
-        Preconditions.checkNotNull(executor, "Executor cannot be null.");
-        return this.getHandle().withEntity(((CraftEntity) executor).getHandle());
     }
 
     // OLD METHODS
