--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -8,6 +_,7 @@
 import java.util.List;
 import java.util.Set;
 import java.util.UUID;
+import net.kyori.adventure.text.minimessage.MiniMessage;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.Tag;
 import net.minecraft.network.chat.Component;
@@ -732,6 +_,11 @@
     @Override
     public void sendMessage(String message) {
 
+    }
+
+    @Override
+    public void sendTranslatedMessage(String messageKey) {
+        this.sendMessage(MiniMessage.miniMessage().deserialize("<red>" + messageKey));
     }
 
     @Override
