--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +_,14 @@
      */
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
+
+    // Varilx start
+    /**
+     * The brand id for varilx.
+     */
+    Key BRAND_VARILX_ID = Key.key("vaxspigot", "vaxspigot");
+    // Varilx end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
