--- a/src/main/java/org/bukkit/craftbukkit/CraftFluid.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftFluid.java
@@ -4,6 +_,7 @@
 import net.minecraft.core.registries.Registries;
 import org.bukkit.Fluid;
 import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
 import org.bukkit.craftbukkit.util.Handleable;
 import org.jetbrains.annotations.NotNull;
 
@@ -12,7 +_,7 @@
     private static int count = 0;
 
     public static Fluid minecraftToBukkit(net.minecraft.world.level.material.Fluid minecraft) {
-        return CraftRegistry.minecraftToBukkit(minecraft, Registries.FLUID);
+        return CraftRegistry.minecraftToBukkit(minecraft, Registries.FLUID, Registry.FLUID);
     }
 
     public static net.minecraft.world.level.material.Fluid bukkitToMinecraft(Fluid bukkit) {
