--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemType.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemType.java
@@ -51,7 +_,7 @@
     }
 
     public static ItemType minecraftToBukkitNew(Item minecraft) {
-        return CraftRegistry.minecraftToBukkit(minecraft, Registries.ITEM);
+        return CraftRegistry.minecraftToBukkit(minecraft, Registries.ITEM, Registry.ITEM);
     }
 
     public static Item bukkitToMinecraftNew(ItemType bukkit) {
