--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
@@ -132,7 +_,7 @@
                 if (menu == null) {
                     return net.minecraft.world.inventory.MenuType.GENERIC_9x3;
                 } else {
-                    return ((CraftMenuType<?, ?>) menu).getHandle();
+                    return ((CraftMenuType<?>) menu).getHandle();
                 }
         }
     }
