--- a/src/test/java/io/papermc/paper/configuration/GlobalConfigTestingBase.java
+++ b/src/test/java/io/papermc/paper/configuration/GlobalConfigTestingBase.java
@@ -1,15 +_,14 @@
 package io.papermc.paper.configuration;
 
-import net.minecraft.core.RegistryAccess;
 import org.spongepowered.configurate.ConfigurationNode;
 import org.spongepowered.configurate.serialize.SerializationException;
 
 public final class GlobalConfigTestingBase {
 
-    public static void setupGlobalConfigForTest(RegistryAccess registryAccess) {
+    public static void setupGlobalConfigForTest() {
         //noinspection ConstantConditions
         if (GlobalConfiguration.get() == null) {
-            ConfigurationNode node = PaperConfigurations.createForTesting(registryAccess);
+            ConfigurationNode node = PaperConfigurations.createForTesting();
             try {
                 GlobalConfiguration globalConfiguration = node.require(GlobalConfiguration.class);
                 GlobalConfiguration.set(globalConfiguration);
