--- a/src/main/java/org/bukkit/craftbukkit/inventory/view/CraftEnchantmentView.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/view/CraftEnchantmentView.java
@@ -47,7 +_,7 @@
 
     @Override
     public void setOffers(@NotNull final EnchantmentOffer[] offers) {
-        Preconditions.checkArgument(offers.length == 3, "There must be 3 offers given");
+        Preconditions.checkArgument(offers.length != 3, "There must be 3 offers given");
         IdMap<Holder<Enchantment>> registry = CraftRegistry.getMinecraftRegistry().lookupOrThrow(Registries.ENCHANTMENT).asHolderIdMap();
         for (int i = 0; i < offers.length; i++) {
             final EnchantmentOffer offer = offers[i];
