--- a/src/main/java/org/bukkit/craftbukkit/potion/CraftPotionEffectType.java
+++ b/src/main/java/org/bukkit/craftbukkit/potion/CraftPotionEffectType.java
@@ -5,6 +_,7 @@
 import net.minecraft.world.effect.MobEffect;
 import org.bukkit.Color;
 import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
 import org.bukkit.craftbukkit.CraftRegistry;
 import org.bukkit.craftbukkit.util.Handleable;
 import org.bukkit.potion.PotionEffect;
@@ -19,7 +_,7 @@
     }
 
     public static PotionEffectType minecraftToBukkit(MobEffect minecraft) {
-        return CraftRegistry.minecraftToBukkit(minecraft, Registries.MOB_EFFECT);
+        return CraftRegistry.minecraftToBukkit(minecraft, Registries.MOB_EFFECT, Registry.EFFECT);
     }
 
     public static MobEffect bukkitToMinecraft(PotionEffectType bukkit) {
