--- a/src/main/java/org/bukkit/craftbukkit/potion/CraftPotionType.java
+++ b/src/main/java/org/bukkit/craftbukkit/potion/CraftPotionType.java
@@ -2,7 +_,6 @@
 
 import com.google.common.base.Preconditions;
 import com.google.common.base.Suppliers;
-import io.papermc.paper.registry.RegistryKey;
 import java.util.List;
 import java.util.Locale;
 import java.util.function.Supplier;
@@ -73,7 +_,7 @@
         if (key == null) return null; // Paper - Fixup NamespacedKey handling
 
         // Now also convert from when keys where saved
-        return CraftRegistry.get(RegistryKey.POTION, key, ApiVersion.CURRENT);
+        return CraftRegistry.get(Registry.POTION, key, ApiVersion.CURRENT);
     }
 
     private final NamespacedKey key;
