From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dario Pranjic <derioreal63@gmail.com>
Date: Mon, 13 Jan 2025 22:08:53 +0100
Subject: [PATCH] Add integration with Varilx CommonAPI


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index e08cf006d6069bce2ddad6e30d5e1f0f7ea9c374..c0be1e3c11a0424670f2fef6a56305b626203d76 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -14,6 +14,7 @@ import java.util.Set;
 import java.util.UUID;
 import java.util.function.Consumer;
 import java.util.logging.Logger;
+import de.varilx.core.common.CommonAPI;
 import org.bukkit.Warning.WarningState;
 import org.bukkit.advancement.Advancement;
 import org.bukkit.block.data.BlockData;
@@ -125,6 +126,10 @@ public final class Bukkit {
         // Paper end
     }
 
+    public static CommonAPI getVarilxCommon() {
+        return getServer().getVarilxCommon();
+    }
+
     /**
      * Gets the name of this server implementation.
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index c60c57c9eb373651b7ed5f00f76fbfb771b24243..63d744d80d5ed6f91f7858043a910cfeac67aaae 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -14,6 +14,7 @@ import java.util.Set;
 import java.util.UUID;
 import java.util.function.Consumer;
 import java.util.logging.Logger;
+import de.varilx.core.common.CommonAPI;
 import org.bukkit.Warning.WarningState;
 import org.bukkit.advancement.Advancement;
 import org.bukkit.block.data.BlockData;
diff --git a/src/main/java/org/bukkit/command/CommandSender.java b/src/main/java/org/bukkit/command/CommandSender.java
index 04e7cd0e4e2d0eb38fb2862ce6688a4470f30d6b..38ffe2fdb4ea63544ec5f26d46351cf6bcd9917f 100644
--- a/src/main/java/org/bukkit/command/CommandSender.java
+++ b/src/main/java/org/bukkit/command/CommandSender.java
@@ -22,6 +22,13 @@ public interface CommandSender extends net.kyori.adventure.audience.Audience, Pe
     @org.jetbrains.annotations.ApiStatus.Obsolete // Paper
     public void sendMessage(@NotNull String message);
 
+    /**
+     * Experimental
+     * @param messageKey
+     */
+    @ApiStatus.Experimental
+    public void sendTranslatedMessage(String messageKey);
+
     /**
      * Sends this sender multiple messages
      *
