--- a/src/main/java/org/bukkit/command/CommandSender.java
+++ b/src/main/java/org/bukkit/command/CommandSender.java
@@ -1,16 +_,40 @@
 package org.bukkit.command;
 
 import java.util.UUID;
+import de.varilx.core.common.translation.Locale;
 import net.kyori.adventure.audience.MessageType;
 import net.kyori.adventure.identity.Identity;
 import net.kyori.adventure.text.Component;
 import org.bukkit.Server;
 import org.bukkit.permissions.Permissible;
+import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public interface CommandSender extends net.kyori.adventure.audience.Audience, Permissible { // Paper
 
+    // Varilx - Start
+
+    /**
+     * This method trys to find the project with the given key
+     * If not found uses the common project
+     * @param key the key
+     * @param args the args for the message
+     * @see CommandSender#sendTranslatedMessage(String, String, Locale, Object...)
+     */
+    public void sendTranslatedMessage(@NotNull String key, @NotNull Object... args);
+
+    /**
+     * This method sends a translated message
+     * @param key the message key
+     * @param project the project where the key is located
+     * @param locale the locale
+     * @param args the args (<1>, <2>, ...)
+     */
+    public void sendTranslatedMessage(@NotNull String key, @NotNull String project, @NotNull Locale locale, @NotNull Object... args);
+
+    // Varilx - End
+
     /**
      * Sends this sender a message
      *
@@ -22,6 +_,7 @@
     @org.jetbrains.annotations.ApiStatus.Obsolete // Paper
     public void sendMessage(@NotNull String message);
 
+
     /**
      * Sends this sender multiple messages
      *
@@ -126,6 +_,7 @@
     // Spigot end
 
     // Paper start
+
     /**
      * Gets the name of this command sender
      *
